### PR TITLE
Create ci.yml

### PR DESCRIPTION
**Pull request by maikelmclauflin**
### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{matrix.goversion}}
+
+    - name: Docker Compose Install
+      uses: KengoTODA/actions-setup-docker-compose@main
```
Comment (diracdeltas) [inline .github/workflows/ci.yml:53]: i'm a little concerned that nobody has audited this repo (given that it only has one open issue). is there a more trustworthy alternative?

Comment (maikelmclauflin) [reply .github/workflows/ci.yml:53]: this one [isbang/compose-action](https://github.com/isbang/compose-action) seems to do basically the same thing but came up first in the search results. not many people bring up with compose. i suppose i could do without? would have to tinker with some docker commands though

Comment (maikelmclauflin) [reply .github/workflows/ci.yml]: used hashes instead of versions

Comment (maikelmclauflin) [reply .github/workflows/ci.yml]: @diracdeltas there is not an official actions/docker-compose from what i can tell though, no

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  pull_request:
+    branches:
+      - master
+      - prod
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (husobee) [inline .github/workflows/golangci-lint.yml:19]: Pin the checkout to the full commit sha
```
    -  uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
```

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
```
Comment (husobee) [inline .github/workflows/golangci-lint.yml:21]: same, pin to full commit sha for golangci-lint-action

**Review by husobee — CHANGES_REQUESTED** [review]
the linting needs to reference full commit, not v2

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (maikelmclauflin) [reply .github/workflows/golangci-lint.yml]: fixed. thanks

File: .github/workflows/golangci-lint.yml (hunk 3 above)
Comment (maikelmclauflin) [reply .github/workflows/golangci-lint.yml]: fixed thanks

**Review by husobee — APPROVED** [review]

**Pull request merged**